src/core/distributions/phylogenetics/tree/coalescent/ConstantPopulationCoalescent.cpp

```diff
@@ -130,42 +196,84 @@ std::vector<double> ConstantPopulationCoalescent::simulateCoalescentAges( size_t
     // Get the rng
     RandomNumberGenerator* rng = GLOBAL_RNG;
     
-    // allocate the vector for the times
-    std::vector<double> coalescent_times = std::vector<double>(n,0.0);
-    
-//    double theta = 1.0 / (2.0*Ne->getValue());
-    double theta = 1.0 / (Ne->getValue());
-    // draw a time for each speciation event condition on the time of the process
-	for (size_t i = 0; i < n; ++i)
+    // retrieve the times of any serially sampled tips
+    std::vector<double> serial_ages;
+    size_t num_taxa_at_present = 0;
+    for (size_t i = 0; i < num_taxa; ++i)
     {
-        double prev_coalescent_time = 0.0;
-        if ( i > 0 ) 
+        double a = taxa[i].getAge();
+        if ( a > 0.0 )
+        {
+            serial_ages.push_back(a);
+        }
+        else
         {
-            prev_coalescent_time = coalescent_times[i-1];
+            ++num_taxa_at_present;
         }
+    }
+    
+    size_t at_serial_age = 0;
+    if (num_taxa_at_present < num_taxa)
+    {
+        std::sort(serial_ages.begin(), serial_ages.end());
+    }
+    
+    // now simulate the ages
+    
+    // allocate the vector for the ages
+    std::vector<double> coalescent_ages = std::vector<double>(n,0.0);
+    
+    // j is the number of active lineages at the current time
+    size_t current_num_lineages = num_taxa_at_present;
+    double theta = Ne->getValue();
```

Comment:
Again, theta = 2*Ne*mu, so why are we writing `theta = Ne`?

Reply:
Because for some arbitrary reasons, the equations of the coalescent use theta as the variable name and define it that way. It's yet another theta ;)